tinygrad/shape/symbolic.py

```diff
@@ -332,10 +332,10 @@ def create_rednode(typ:Type[RedNode], nodes:List[Node]):
   return create_node(ret)
 
 @functools.lru_cache(maxsize=None)
-def sym_rename(s) -> str: return f"s{sym_rename.cache_info().currsize}"
-def sym_render(a: Union[Node, int], ops=None, ctx=None) -> str: return str(a) if isinstance(a, int) else a.render(ops, ctx)
-def sym_infer(a: Union[Node, int], var_vals: Dict[Variable, int]) -> int:
-  if isinstance(a, (int, float)): return a
+def sym_rename(s:sint) -> str: return f"s{sym_rename.cache_info().currsize}"
+def sym_render(a:sint, ops=None, ctx=None) -> str: return str(a) if isinstance(a, int) else a.render(ops, ctx)
+def sym_infer(a:sint, var_vals: Dict[Variable, int]) -> int:
+  if isinstance(a, int): return a
```

Comment:
Err...what if it's a float?

Reply:
Can you add a test to catch this regression? Look at the NOLOCALS stuff for when it can be a float

Reply:
You are right, if it is a float it will break. Also it would mean, the types were wrong before.

I will revert this, add a todo and and create a separate PR.